Data/Map/Strict/Internal.hs

```diff
@@ -887,9 +887,20 @@ atKeyIdentity k f t = Identity $ atKeyPlain Strict k (coerce f) t
 
 #if __GLASGOW_HASKELL__ >= 800
 updateAt :: HasCallStack => (k -> a -> Maybe a) -> Int -> Map k a -> Map k a
+updateAt = go where
```

Comment:
Does this actually give the right call stacks? Thinking about this more carefully, I suspect you want to check if the operation will succeed, and only then call a `go` function that takes no call stack. The alternative would be to call the `go` function using `withFrozenCallStack`, which I suspect will be more expensive. Benchmarks may tell.

Reply:
My tests show that, by default, functions like `go` do not have the `HasCallStack` constraint unless it is explicit in their type signatures.  Therefore, this seems to give the right call stack.  There's no need to check if the operation succeeds or not first, and because `go` doesn't have the constraint, there's no need for `withFrozenCallStack`.